### DISCUSSION TITLE
No json schema in Pydantic v2

### DISCUSSION THREAD

**Discussion by deivydaslipskis** [pydantic/pydantic#7216] [Pydantic V2]
Hi, I am migrating to Pydantic v2 and I have noticed that there is no `BaseModel.schema` (https://docs.pydantic.dev/1.10/usage/schema/) alternative in Pydantic v2.

I have two different lines of code to generate schema as `dict` and as `json`:

```
json_schema = model.schema_json(indent=2, by_alias=True)
dict_schema = model.schema(by_alias=True)
```

For both `schema_json` and `schema` editor advises me to use `model_json_schema` instead. Although it is used for returning `dict[str, Any]`.

Is there a reason there is no json schema or was this missed?

**Comment by harunyasar** [comment]
You could use `json.dumps(model.model_json_schema())`.

**Reply by deivydaslipskis** [reply]
In this case, should I set `by_alias` inside `model_json_schema`? And set indent in the `dumps` function?
`json.dumps(model.model_json_schema(by_alias=True), indent=2)`

**Reply by harunyasar** [reply]
I guess so 😊
